packages/algob/src/internal/deployer.ts

```diff
@@ -1054,4 +1105,44 @@ export class DeployerRunMode extends DeployerBasicMode implements Deployer {
 			scTmplParams
 		);
 	}
+	/**
+	 * Execute single transaction or group of transactions (atomic transaction)
+	 * executes `ExecParams` or `Transaction` Object, SDK Transaction object passed to this function
+	 * will be signed and sent to network. User can use SDK functions to create transactions.
+	 * Note: If passing transaction object a signer/s must be provided.
+	 * @param deployer Deployer
+	 * @param transactionParam transaction parameters or atomic transaction parameters
+	 * https://github.com/scale-it/algo-builder/blob/docs/docs/guide/execute-transaction.md
+	 * or TransactionAndSign object(SDK transaction object and signer parameters)
+	 */
+	async executeTx(
+		transactions: wtypes.ExecParams[] | wtypes.TransactionAndSign[]
+	): Promise<ConfirmedTxInfo[]> {
+		let isSDK = false;
+		let signedTxn;
+		if (transactions.length === 0) {
+			throw new BuilderError(ERRORS.GENERAL.EXECPARAMS_LENGTH_ERROR);
+		}
+		if (wtypes.isSDKTransactionAndSign(transactions[0])) {
+			signedTxn = signTransactions(transactions as wtypes.TransactionAndSign[]);
+			isSDK = true;
+		}
+
+		if (isSDK && signedTxn) {
+			await this.sendAndWait(signedTxn);
+			return await this.getReceiptTxns(
+				(transactions as wtypes.TransactionAndSign[]).map((txn) => txn.transaction)
+			);
+		}
+
+		const execParams = transactions as wtypes.ExecParams[];
+
+		this.assertCPNotDeleted(execParams);
+		const txIdxMap = new Map<number, [string, wtypes.ASADef]>();
+		const [txns, signedTxns] = await makeAndSignTx(this, execParams, txIdxMap);
+		await this.sendAndWait(signedTxns);
+		const confirmedTx = await this.getReceiptTxns(txns);
+		console.debug(confirmedTx);
```

Comment:
this lines are repeated, can we move it to a common private function?

Reply:
I am not sure what you mean by a private function. It cannot be moved 'up' because the base class does not implement the Deployer interface. To not repeat the code I moved it back as a function and just invoke from inside of the executeTx method in Deployer class. It will make the old code compatible but also allows users to use it as a deployer method now. LMK what do you think about it

Reply:
I was thinking about some function in the file, which will take deployer as an argument. But if it's too hacky then we can keep as is.